### ISSUE TITLE
ftplib.error_perm: 550 No such file or directory

### ISSUE THREAD

**Issue by Alxzu** [a1ex4/ownfoil#23]
Hi, first of all thanks for sharing this amazing project. I'm having an issue trying to get my saves from the switch. 

FTP server is running smoothly so no issues on that matter. It seems like an error when trying to run the backup script.

```
Starting ownfoil
2023-05-04 17:49:28,368 - INFO apscheduler.scheduler: Adding job tentatively -- it will be properly scheduled when the scheduler starts
2023-05-04 17:49:28,368 - INFO apscheduler.scheduler: Adding job tentatively -- it will be properly scheduled when the scheduler starts
2023-05-04 17:49:28,377 - INFO apscheduler.scheduler: Added job "Generate shop" to job store "default"
2023-05-04 17:49:28,377 - INFO apscheduler.scheduler: Added job "Backup saves" to job store "default"
2023-05-04 17:49:28,377 - INFO apscheduler.scheduler: Scheduler started
2023-05-04 17:49:28,379 - INFO apscheduler.executors.default: Running job "Generate shop (trigger: interval[0:05:00], next run at: 2023-05-04 17:49:28 UTC)" (scheduled at 2023-05-04 17:49:28.078178+00:00)
2023-05-04 17:49:28,380 - INFO apscheduler.executors.default: Running job "Backup saves (trigger: interval[0:05:00], next run at: 2023-05-04 17:54:28 UTC)" (scheduled at 2023-05-04 17:49:28.368518+00:00)
2023-05-04 17:49:28,383 - INFO gen_shop: Found 0 directories, 0 game/save files
2023-05-04 17:49:28,389 - INFO gen_shop: Successfully wrote /games/shop.json
2023-05-04 17:49:28,390 - INFO gen_shop: Successfully wrote /games/shop.tfl
2023-05-04 17:49:28,391 - INFO apscheduler.executors.default: Job "Generate shop (trigger: interval[0:05:00], next run at: 2023-05-04 17:54:28 UTC)" executed successfully
2023-05-04 17:49:28,421 - INFO save_manager: Successfully connected to Switch device on host 192.168.0.105.
2023-05-04 17:49:28,421 - INFO save_manager: Retrieving saves from /switch/tinfoil/saves/common to /games/Saves/Tinfoil
2023-05-04 17:49:28,461 - ERROR apscheduler.executors.default: Job "Backup saves (trigger: interval[0:05:00], next run at: 2023-05-04 17:54:28 UTC)" raised an exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/apscheduler/executors/base.py", line 125, in run_job
    retval = job.func(*job.args, **job.kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backup_saves.py", line 195, in backup_saves
    switch_ftp.retrieve_saves(config['root_dir'] + '/' + folder['local'], folder['remote'])
  File "/app/backup_saves.py", line 106, in retrieve_saves
    dirs = self.get_dirs(remote_folder)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/backup_saves.py", line 94, in get_dirs
    res = self.ftp.retrlines(f'NLST {path}', lambda x: dirs.append(x))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ftplib.py", line 462, in retrlines
    with self.transfercmd(cmd) as conn, \
         ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ftplib.py", line 393, in transfercmd
    return self.ntransfercmd(cmd, rest)[0]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ftplib.py", line 359, in ntransfercmd
    resp = self.sendcmd(cmd)
           ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ftplib.py", line 281, in sendcmd
    return self.getresp()
           ^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/ftplib.py", line 254, in getresp
    raise error_perm(resp)
ftplib.error_perm: 550 No such file or directory
```

Attaching my docker compose. `PUID` & `PGID` are ok

![Screen Shot 2023-05-04 at 14 56 02](https://user-images.githubusercontent.com/8084303/236288915-d562a23f-991b-40c8-9301-bf2cce5e60f8.png)

**Comment by CortezSMz** [comment]
The path "/switch/tinfoil/saves/common" probably does not exists on your switch.

You can check tinfoil's save location by going to: Tinfoil > Options > and scrolling down to "Game Save Backup"

**Comment by irishj** [comment]
I have the same issue, I'm using JKSV for saves and have omitted the line for tinfoil saves in the shop_config.toml file

Running a FTP server manually on the switch, I can see the connection attempt by the shop and an error.

[RESPONSE] 550 Not a file

This seems to be correct, as checking the structure of the JKSV folder, it contains directories for each game, then the username for the save, then the actual saves.  In certain cases, like with "Wolfenstein  II  The New Colossus", the structure is as follows.

DIR = Wolfenstein  II  The New Colossus
- SUBDIR - username
-- SUBDIR - GAME-AUTOSAVE0-SLOT-0
-- SUBDIR - GAME-TRANSITION-SLOT-0
-- SUBDIR - PROFILE
-- SUBDIR - PROFILE-SLOT-0

It appears the sFTP client in the shop is expecting a file, but instead is finding a directory and thus returns the "550 Not A File" error response.

**Comment by CortezSMz** [comment]
> I have the same issue, I'm using JKSV for saves and have omitted the line for tinfoil saves in the shop_config.toml file
> 
> Running a FTP server manually on the switch, I can see the connection attempt by the shop and an error.
> 
> [COMMAND] SIZE /JKSV/Wolfenstein II The New Colossus/USERID - 2023.05.10 @ 18.40.08 [RESPONSE] 550 Not a file
> 
> This seems to be correct, as checking the structure of the JKSV folder, it contains directories for each game, then the username for the save, then the actual saves. In certain cases, like with "Wolfenstein II The New Colossus", the structure is as follows.
> 
> DIR = Wolfenstein II The New Colossus
> 
> * SUBDIR - username
>   -- SUBDIR - GAME-AUTOSAVE0-SLOT-0
>   -- SUBDIR - GAME-TRANSITION-SLOT-0
>   -- SUBDIR - PROFILE
>   -- SUBDIR - PROFILE-SLOT-0
> 
> It appears the FTP client in the shop is expecting a file, but instead is finding a directory and thus returns the "550 Not A File" error response.
> 
> Tried deleting that one folder in case it was an issue specific to that game, but happened with another after that.

This should be fixed using the `a1ex4/ownfoil:develop` docker image. If possible can you test and confirm that?

**Comment by irishj** [comment]
Checked develop branch and can confirm it's working there.  Thanks for the quick update !

**Comment by a1ex4** [comment]
Thanks @CortezSMz, I've released [version 1.2.3](https://github.com/a1ex4/ownfoil/releases/tag/1.2.3) so people won't experience the issue anymore.
